staging/src/k8s.io/apiserver/pkg/storage/etcd3/watcher_test.go

```diff
@@ -118,27 +119,27 @@ func testWatch(t *testing.T, recursive bool) {
 					expectObj = prevObj
 					expectObj.ResourceVersion = out.ResourceVersion
 				}
-				testCheckResult(t, i, watchTest.watchType, w, expectObj)
+				testCheckResult(t, i, watchTest.watchType, aw, expectObj)
 			}
 			prevObj = out
 		}
-		w.Stop()
-		testCheckStop(t, i, w)
+		aw.Stop()
+		testCheckStop(t, i, aw)
 	}
 }
 
 func TestDeleteTriggerWatch(t *testing.T) {
 	ctx, store, cluster := testSetup(t)
 	defer cluster.Terminate(t)
 	key, storedObj := testPropogateStore(ctx, t, store, &example.Pod{ObjectMeta: metav1.ObjectMeta{Name: "foo"}})
-	w, err := store.Watch(ctx, key, storedObj.ResourceVersion, storage.Everything)
-	if err != nil {
-		t.Fatalf("Watch failed: %v", err)
+	aw := store.Watch(ctx, key, storedObj.ResourceVersion, storage.Everything)
+	if aw.GetFirstError() != nil {
```

Comment:
maybe allerrors?

Reply:
For error case, we need to implement retry. It can be done in watch.go. For test, right now, there are too many places to change. Use FirstError for now.